kernel/aster-nix/src/syscall/arch/x86.rs

```diff
@@ -0,0 +1,251 @@
+    sync::sys_sync,
+    tgkill::sys_tgkill,
+    time::sys_time,
+    truncate::{sys_ftruncate, sys_truncate},
+    umask::sys_umask,
+    uname::sys_uname,
+    unlink::{sys_unlink, sys_unlinkat},
+    utimens::sys_utimensat,
+    wait4::sys_wait4,
+    waitid::sys_waitid,
+    write::sys_write,
+    writev::sys_writev,
+};
+
+impl_syscall_nums_and_dispatch_fn! {
+    SYS_READ = 0               => sys_read(args[..3]);
+    SYS_WRITE = 1              => sys_write(args[..3]);
+    SYS_OPEN = 2               => sys_open(args[..3]);
+    SYS_CLOSE = 3              => sys_close(args[..1]);
+    SYS_STAT = 4               => sys_stat(args[..2]);
+    SYS_FSTAT = 5              => sys_fstat(args[..2]);
+    SYS_LSTAT = 6              => sys_lstat(args[..2]);
+    SYS_POLL = 7               => sys_poll(args[..3]);
+    SYS_LSEEK = 8              => sys_lseek(args[..3]);
+    SYS_MMAP = 9               => sys_mmap(args[..6]);
+    SYS_MPROTECT = 10          => sys_mprotect(args[..3]);
+    SYS_MUNMAP = 11            => sys_munmap(args[..2]);
+    SYS_BRK = 12               => sys_brk(args[..1]);
+    SYS_RT_SIGACTION = 13      => sys_rt_sigaction(args[..4]);
+    SYS_RT_SIGPROCMASK = 14    => sys_rt_sigprocmask(args[..4]);
+    SYS_RT_SIGRETURN = 15      => sys_rt_sigreturn(args[..0], context);
+    SYS_IOCTL = 16             => sys_ioctl(args[..3]);
+    SYS_PREAD64 = 17           => sys_pread64(args[..4]);
+    SYS_WRITEV = 20            => sys_writev(args[..3]);
+    SYS_ACCESS = 21            => sys_access(args[..2]);
+    SYS_PIPE = 22              => sys_pipe(args[..1]);
+    SYS_SELECT = 23            => sys_select(args[..5]);
+    SYS_SCHED_YIELD = 24       => sys_sched_yield(args[..0]);
+    SYS_MADVISE = 28           => sys_madvise(args[..3]);
+    SYS_DUP = 32               => sys_dup(args[..1]);
+    SYS_DUP2 = 33              => sys_dup2(args[..2]);
+    SYS_PAUSE = 34             => sys_pause(args[..0]);
+    SYS_NANOSLEEP = 35         => sys_nanosleep(args[..2]);
+    SYS_ALARM = 37             => sys_alarm(args[..1]);
+    SYS_GETPID = 39            => sys_getpid(args[..0]);
+    SYS_SOCKET = 41            => sys_socket(args[..3]);
+    SYS_CONNECT = 42           => sys_connect(args[..3]);
+    SYS_ACCEPT = 43            => sys_accept(args[..3]);
+    SYS_SENDTO = 44            => sys_sendto(args[..6]);
+    SYS_RECVFROM = 45          => sys_recvfrom(args[..6]);
+    SYS_SHUTDOWN = 48          => sys_shutdown(args[..2]);
+    SYS_BIND = 49              => sys_bind(args[..3]);
+    SYS_LISTEN = 50            => sys_listen(args[..2]);
+    SYS_GETSOCKNAME = 51       => sys_getsockname(args[..3]);
+    SYS_GETPEERNAME = 52       => sys_getpeername(args[..3]);
+    SYS_SOCKETPAIR = 53        => sys_socketpair(args[..4]);
+    SYS_SETSOCKOPT = 54        => sys_setsockopt(args[..5]);
+    SYS_GETSOCKOPT = 55        => sys_getsockopt(args[..5]);
+    SYS_CLONE = 56             => sys_clone(args[..5], context);
```

Comment:
In Rust, we prefer immutable variables/references over mutable ones. So I suggest differentiate between immutable and mutable references to `CpuContext`. Can we support the following syntax?

```rust
    SYS_CLONE = 56             => sys_clone(args[..5], &context);
    SYS_RT_SIGRETURN = 15      => sys_rt_sigreturn(args[..0], &mut context);

```